src/components/fieldset/fieldset.test.js

```diff
@@ -0,0 +1,212 @@
+import * as React from "react";
+import Fieldset from "./fieldset.component";
+import legendPreview from "../../../cypress/locators/fieldset";
+import Textbox from "../textbox/textbox.component";
+import Checkbox from "../checkbox/checkbox.component";
+import Form from "../form/form.component";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import { getDataElementByValue } from "../../../cypress/locators/index";
+import { positionOfElement } from "../../../cypress/support/helper";
+
+const specialCharacters = [
+  "legend",
+  "mp150ú¿¡üßä",
+  "!@#$%^*()_+-=~[];:.,?{}&\"'<>",
+];
+
+const FieldsetComponent = ({ ...props }) => {
+  return (
+    <div>
+      <Fieldset legend="Fieldset" {...props}>
+        <Textbox
+          label="First Name"
+          labelInline
+          labelAlign="right"
+          labelWidth={30}
+        />
+        <Textbox
+          label="Last Name"
+          labelInline
+          labelAlign="right"
+          labelWidth={30}
+        />
+        <Textbox
+          label="Address"
+          labelInline
+          labelAlign="right"
+          labelWidth={30}
+        />
+        <Checkbox
+          label="Checkbox"
+          labelAlign="right"
+          labelWidth={30}
+          labelSpacing={2}
+          reverse
+        />
+        <Textbox label="City" labelInline labelAlign="right" labelWidth={30} />
+        <Textbox
+          label="Country"
+          labelInline
+          labelAlign="right"
+          labelWidth={30}
+        />
+        <Textbox
+          label="Telephone"
+          labelInline
+          labelAlign="right"
+          labelWidth={30}
+        />
+      </Fieldset>
+    </div>
+  );
+};
+
+context("Testing Fieldset component", () => {
+  describe("should render Fieldset component", () => {
+    it.each(specialCharacters)(
+      "should verify Fieldset preview text is %s",
+      (chars) => {
+        CypressMountWithProviders(<FieldsetComponent legend={chars} />);
+
+        legendPreview().should("have.text", chars);
+      }
+    );
+
+    it("should verify Fieldset preview is not displayed", () => {
+      CypressMountWithProviders(<FieldsetComponent legend="" />);
+
+      legendPreview().should("not.exist");
+    });
+
+    it.each([
+      [true, 36, 39.984375, 78.1875],
+      [false, 18, 76.859375, 919],
+    ])(
+      "should verify Fieldset preview is not displayed",
```

Comment:
The test name appears to be the same as the test above but shouldn't it specify that it relates to verifying that the fields are inline or not?

Reply:
Yes it should. The perils of copying and pasting!